src/graphql/schema/uuid/abstract-repository/utils.ts

```diff
@@ -43,6 +44,29 @@ export function createRepositoryResolvers<
       if (!entity.currentRevisionId) return null
       return dataSources.serlo.getUuid<R>({ id: entity.currentRevisionId })
     },
+    async revisions(entity, cursorPayload, { dataSources }) {
+      const revsFromSerlo = await Promise.all(
+        entity.revisionIds.map((id) => {
+          return dataSources.serlo.getUuid<R>({ id })
+        })
+      )
+      const revs = revsFromSerlo.filter(isNotNil)
+      const filteredRevs = cursorPayload.unrevised
+        ? revs.filter(
+            (rev) =>
+              (entity.currentRevisionId === null ||
+                rev.id > entity.currentRevisionId) &&
+              !rev.trashed
+          )
+        : revs
```

Comment:
same here:
```suggestion
    .filter(cursorPayload.unrevised ? (rev) =>
              (entity.currentRevisionId === null ||
                rev.id > entity.currentRevisionId) &&
              !rev.trashed
          ) : (revision) => true)
```
It's probably more readable if you move the `cursorPayload.unrevised` query into the filter itself, so something like:
```typescript
(revision) => {
  if (cursorPayload.unrevised === null) return true // Don't filter at all, see comment below
  // Do something depending on whether cursorPayload.unrevised is true or false
  return (entity.currentRevisionId === null ||
                  rev.id > entity.currentRevisionId) &&
                !rev.trashed
  ) // This line can probably be improved too
}

Reply:
Another thing: the semantic isn't what we want at the moment. We are interested in pretty much three cases here:
- Give me all revisions (e.g. for the history https://de.serlo.org/entity/repository/history/1855)
- Give me all unrevised revisions (e.g. so that we can show if there are any unrevised revisions)
- Give me all revised revisions (e.g. to show only the accepted revisions and/or know how often an article got revised).

We can handle that by being mindful whether `cursorPayload.unrevised` is defined or not:
- `null`: the consumer didn't specify any filter. So we return all revisions
- `true`: the consumer specified that they only want the unrevised revisions
- `false` the consumer specified that they only want the revised revisions

We did something similar for the notifications, see notification/resolvers.ts